### PR TITLE
RUMM3387 Align v2 Public API

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Rename configuration fields to align with Browser and Android.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [x] Run integration tests
- [x] Run smoke tests

### CODE REVIEW

File: Datadog/Example/Debugging/DebugLoggingViewController.swift [hunk 1]
```diff
@@ -94,7 +94,8 @@ class DebugLoggingViewController: UIViewController {
             return Logger.create(
                 with: Logger.Configuration(
                     name: "stress-logger-\(index)",
-                    sendNetworkInfo: true
+                    networkInfoEnabled
+                    : true
```
Comment (ncreated) [inline Datadog/Example/Debugging/DebugLoggingViewController.swift:98]: /nit 
```suggestion
                    networkInfoEnabled: true
```

File: DatadogCore/Sources/Datadog.swift [hunk 2]
```diff
@@ -237,37 +242,70 @@ public struct Datadog {
...
+    ///         with: RUM.Configuration(applicationID: "<application>"),
+    ///         in: core
+    ///     )
+    ///
     /// - Parameters:
     ///   - configuration: the SDK configuration.
     ///   - trackingConsent: the initial state of the Data Tracking Consent given by the user of the app.
     ///   - instanceName:   The core instance name. This value will be used for data persistency and should be
     ///                     stable between application runs.
+    @discardableResult
```
Comment (ncreated) [inline DatadogCore/Sources/Datadog.swift:281]: 👌

**Review by ncreated — APPROVED** [review]
🎯

File: DatadogCore/Sources/Datadog.swift [hunk 3]
```diff
@@ -237,37 +242,70 @@ public struct Datadog {
...
+    ///     let core = Datadog.initialize(
+    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
+    ///         trackingConsent: .pending,
+    ///         instanceName: "my-instance"
+    ///     )
+    ///
+    ///     RUM.enable(
+    ///         with: RUM.Configuration(applicationID: "<application>"),
+    ///         in: core
+    ///     )
```
Comment (ganeshnj) [inline DatadogCore/Sources/Datadog.swift:274]:
```suggestion
    ///    ```
    ///     Datadog.initialize(
    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
    ///         trackingConsent: .pending
    ///     )
    ///    ```
    ///
    /// Once Datadog SDK is initialized, you can enable products, such as RUM:
    ///    ```
    ///     RUM.enable(
    ///         with: RUM.Configuration(applicationID: "<application>")
    ///     )
    ///    ```
    /// It is possible to initialize multiple instances of the SDK, associating them with a name.
    /// Many methods of the SDK can optionally take a SDK instance as an argument. If not provided,
    /// the call will be associated with the default (nameless) SDK instance.
    ///
    /// To use a secondary instance of the SDK, provide a name to the ``initialize`` method
    /// and use the returned instance to enable products:
    ///
    ///    ```
    ///     let core = Datadog.initialize(
    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
    ///         trackingConsent: .pending,
    ///         instanceName: "my-instance"
    ///     )
    ///
    ///     RUM.enable(
    ///         with: RUM.Configuration(applicationID: "<application>"),
    ///         in: core
    ///     )
    ///    ```
```

Comment (ganeshnj) [reply DatadogCore/Sources/Datadog.swift:274]: markdown syntax will render nice in our public API docs.

**Review by ganeshnj — APPROVED** [review]

File: DatadogCore/Sources/Datadog.swift (hunk 3 above)
Comment (ganeshnj) [reply DatadogCore/Sources/Datadog.swift:274]: https://developer.apple.com/documentation/xcode/formatting-your-documentation-content#Add-code-listings

**Pull request merged**